runtime/src/main/cpp/Exceptions.cpp

```diff
@@ -291,16 +299,32 @@ static void KonanTerminateHandler() {
   }
 }
 
+#if KONAN_WINDOWS
+#include <windows.h>
+
+LONG WINAPI KonanUnhandledExceptionFilter(EXCEPTION_POINTERS* ExceptionInfo) {
+  if (currentException != nullptr) {
+    auto e = currentException;
```

Comment:
What happens if the following sequence of events happens?
1.  Kotlin exception is thrown and caught.
2.  Non-Kotlin exception is thrown and not caught, causing `KonanUnhandledExceptionFilter` to be called.